e2e/dractions/retry.go

```diff
@@ -143,3 +146,75 @@ func waitDRPCProgression(
 		}
 	}
 }
+
+func addDRPCAnnotation(ctx types.TestContext, namespace, drpcName string) error {
+	log := ctx.Logger()
+
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		drpc, err := getDRPC(ctx, namespace, drpcName)
+		if err != nil {
+			return err
+		}
+
+		if drpc.Annotations == nil {
+			drpc.Annotations = make(map[string]string)
+		}
+
+		drpc.Annotations[drpcDoNotDeletePVC] = "true"
+
+		if err := updateDRPC(ctx, drpc); err != nil {
+			return err
+		}
+
+		log.Debugf("Annotated drpc \"%s/%s\" with \"%s: %s\" in cluster %q",
+			namespace, drpcName, drpcDoNotDeletePVC,
+			drpc.Annotations[drpcDoNotDeletePVC], ctx.Env().Hub.Name)
+
+		return nil
+	})
+}
+
+func waitForDRPCAnnotationPropagation(ctx types.TestContext, namespace, drpcName string) error {
+	log := ctx.Logger()
+	start := time.Now()
+
+	log.Debugf("Waiting until drpc \"%s/%s\" annotation %q is propagated to primary vrg",
+		namespace, drpcName, drpcDoNotDeletePVC)
+
+	for {
+		drpc, err := getDRPC(ctx, namespace, drpcName)
+		if err != nil {
+			return err
+		}
+
+		preferredClusterName := drpc.Spec.PreferredCluster
+
+		preferredCluster, err := ctx.Env().GetCluster(preferredClusterName)
+		if err != nil {
+			return err
+		}
+
+		vrgName := drpcName
+
+		vrg, err := getVRG(ctx, preferredCluster, ctx.AppNamespace(), vrgName)
+		if err != nil {
+			return err
+		}
+
+		if vrg.Annotations != nil {
+			if value, exists := vrg.Annotations[drpcDoNotDeletePVC]; exists && value == "true" {
+				elapsed := time.Since(start)
+				log.Debugf("DRPC \"%s/%s\" annotation \"%s: %s\" propagated to primary vrg \"%s/%s\" in cluster %q in %.3f seconds",
+					namespace, drpcName, drpcDoNotDeletePVC, drpc.Annotations[drpcDoNotDeletePVC], ctx.AppNamespace(),
```

Comment:
We just validated that the value is "true" so we don't need to get it again from the map. Showing the expected annotation in the log is nice.